### PR TITLE
Webauthn Authentication Module

### PR DESCRIPTION
**Pull request by maximthomas**

### CODE REVIEW

File: openam-authentication/openam-auth-webauthn/src/main/resources/amAuthWebAuthn.properties [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+org.openidentityplatform.openam.authentication.modules.webauthn.WebAuthnRegistration.attestation=Attestation Type
+org.openidentityplatform.openam.authentication.modules.webauthn.WebAuthnRegistration.attestation.none=None
+org.openidentityplatform.openam.authentication.modules.webauthn.WebAuthnRegistration.attestation.indirect=Indirect
+org.openidentityplatform.openam.authentication.modules.webauthn.WebAuthnRegistration.authType=Authenticator Type
+org.openidentityplatform.openam.authentication.modules.webauthn.WebAuthnRegistration.timeout=Timeout
+org.openidentityplatform.openam.authentication.modules.webauthn.WebAuthnRegistration.userAttribute=User attribute to store Public Keys
+
+org.openidentityplatform.openam.authentication.modules.webauthn.WebAuthnAuthentication.description=WebAuthn Authentication
+org.openidentityplatform.openam.authentication.modules.webauthn.WebAuthnAuthentication.timeout=Timeout
+org.openidentityplatform.openam.authentication.modules.webauthn.WebAuthnAuthentication.authlevel=Auth Level
```
Comment (vharseko) [inline openam-authentication/openam-auth-webauthn/src/main/resources/amAuthWebAuthn.properties:14]: duplicate authlevel

File: openam-authentication/openam-auth-webauthn/src/main/resources/amAuthWebAuthn.properties [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+org.openidentityplatform.openam.authentication.modules.webauthn.WebAuthnRegistration.description=WebAuthn Registration
+org.openidentityplatform.openam.authentication.modules.webauthn.WebAuthnRegistration.authlevel=Auth Level
+org.openidentityplatform.openam.authentication.modules.webauthn.WebAuthnRegistration.attestation=Attestation Type
+org.openidentityplatform.openam.authentication.modules.webauthn.WebAuthnRegistration.attestation.none=None
+org.openidentityplatform.openam.authentication.modules.webauthn.WebAuthnRegistration.attestation.indirect=Indirect
+org.openidentityplatform.openam.authentication.modules.webauthn.WebAuthnRegistration.authType=Authenticator Type
+org.openidentityplatform.openam.authentication.modules.webauthn.WebAuthnRegistration.timeout=Timeout
+org.openidentityplatform.openam.authentication.modules.webauthn.WebAuthnRegistration.userAttribute=User attribute to store Public Keys
+
+org.openidentityplatform.openam.authentication.modules.webauthn.WebAuthnAuthentication.description=WebAuthn Authentication
```
Comment (vharseko) [inline openam-authentication/openam-auth-webauthn/src/main/resources/amAuthWebAuthn.properties:12]: duplicate descriptions

File: openam-authentication/openam-auth-webauthn/src/main/resources/amAuthWebAuthn.properties [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+org.openidentityplatform.openam.authentication.modules.webauthn.WebAuthnAuthentication.timeout=Timeout
+org.openidentityplatform.openam.authentication.modules.webauthn.WebAuthnAuthentication.authlevel=Auth Level
+org.openidentityplatform.openam.authentication.modules.webauthn.WebAuthnAuthentication.userAttribute=User attribute to retrieve Public Keys
```
Comment (vharseko) [inline openam-authentication/openam-auth-webauthn/src/main/resources/amAuthWebAuthn.properties:15]: duplicate

File: openam-authentication/openam-auth-webauthn/src/main/java/org/openidentityplatform/openam/authentication/modules/webauthn/WebAuthnAuthentication.java [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+		
+		
+		AuthenticatorData<?> authenticatorData = webAuthnAuthenticationProcessor.processCredentials(getHttpServletRequest(), id, authenticatorDataStr, 
+				clientDataJSONStr, signatureStr, userHandleStr, authenticators);
+		
+		if(authenticatorData == null) {
+			logger.warn("processCredentials: authenticator data with id {} not found for identity: {}", id, username);
+			throw new InvalidPasswordException("authenticator not found");
+		}
+		return ISAuthConstants.LOGIN_SUCCEED;
```
Comment (vharseko) [inline openam-authentication/openam-auth-webauthn/src/main/java/org/openidentityplatform/openam/authentication/modules/webauthn/WebAuthnAuthentication.java:186]: don't setAuthLevel

**Review by vharseko — CHANGES_REQUESTED** [review]

**Comment by maximthomas** [comment]
@vharseko see bd469f4

**Comment by maximthomas** [comment]
Added local development notes: 8e58d69 e2937f1

File: openam-authentication/openam-auth-webauthn/README.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+
+## WebAuthn Authentication
+
+User authentication, using previously registered public key.
+
+### Settings
+| Setting | Description |
+|--------|-------------|
+|**Auth Level**| Modlue Authentication Level|
```
Comment (vharseko) [inline openam-authentication/openam-auth-webauthn/README.md:36]: Module ?

Comment (maximthomas) [reply openam-authentication/openam-auth-webauthn/README.md:36]: Headings at 13 line and 29 line

**Pull request merged**